libraries/apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/introspection/introspection_to_schema.kt

```diff
@@ -274,6 +280,50 @@ private class GQLDocumentBuilder(private val introspectionSchema: IntrospectionS
         directives = emptyList()
     )
   }
+
+  private fun IntrospectionSchema.Schema.Directive.toGQLDirectiveDefinition(): GQLDirectiveDefinition {
+    return GQLDirectiveDefinition(
+        sourceLocation = sourceLocation,
+        description = description,
+        name = name,
+        arguments = args.map { it.toGQLInputValueDefinition() },
+        locations = locations.map { it.toGQLDirectiveLocations() },
+        repeatable = isRepeatable,
+    )
+  }
+
+  private fun IntrospectionSchema.Schema.Directive.Argument.toGQLInputValueDefinition(): GQLInputValueDefinition {
```

Comment:
Note for 4.x: I'm almost sure `Directive.Argument` is the same as `Field.Argument` (and since we're going to use generated model, we can certainly share a fragment there)